opc/resource_orchestrated_instance_test.go

```diff
@@ -0,0 +1,528 @@
+					testAccCheckOrchestrationExists,
+					resource.TestCheckResourceAttrSet(resName, "instance.0.id"),
+				),
+			},
+			{
+				Config: testAccOrchestrationInactive(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOrchestrationExists,
+				),
+			},
+		},
+	})
+}
+
+func TestAccOPCOrchestratedInstance_inactiveToActive(t *testing.T) {
+	ri := acctest.RandInt()
+	resName := "opc_compute_orchestrated_instance.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckOrchestrationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccOrchestrationInactive(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOrchestrationExists,
+				),
+			},
+			{
+				Config: testAccOrchestrationBasic(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOrchestrationExists,
+					resource.TestCheckResourceAttrSet(resName, "instance.0.id"),
+				),
+			},
+		},
+	})
+}
+
+func TestAccOPCOrchestratedInstance_activeToSuspend(t *testing.T) {
+	ri := acctest.RandInt()
+	resName := "opc_compute_orchestrated_instance.test"
+
+	resource.Test(t, resource.TestCase{
+		PreCheck:     func() { testAccPreCheck(t) },
+		Providers:    testAccProviders,
+		CheckDestroy: testAccCheckOrchestrationDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: testAccOrchestrationBasic(ri),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckOrchestrationExists,
+					resource.TestCheckResourceAttrSet(resName, "instance.0.id"),
+				),
+			},
+		},
+	})
+}
+
+func testAccCheckOrchestrationExists(s *terraform.State) error {
+	client := testAccProvider.Meta().(*OPCClient).computeClient.Orchestrations()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opc_compute_orchestration" {
+			continue
+		}
+
+		input := compute.GetOrchestrationInput{
+			Name: rs.Primary.Attributes["name"],
+		}
+		if _, err := client.GetOrchestration(&input); err != nil {
+			return fmt.Errorf("Error retrieving state of Orchestration %s: %s", input.Name, err)
+		}
+	}
+
+	return nil
+}
+
+func testAccCheckOrchestrationDestroy(s *terraform.State) error {
+	client := testAccProvider.Meta().(*OPCClient).computeClient.Orchestrations()
+
+	for _, rs := range s.RootModule().Resources {
+		if rs.Type != "opc_compute_orchestrated_instance" {
+			continue
+		}
+
+		input := compute.GetOrchestrationInput{
+			Name: rs.Primary.Attributes["name"],
+		}
+		if info, err := client.GetOrchestration(&input); err == nil {
+			return fmt.Errorf("Orchestration %s still exists: %#v", input.Name, info)
+		}
+	}
+
+	return nil
+}
+
+func testAccGetInstanceAttr(instance, attr string) string {
+	return fmt.Sprintf("%s.%s", instance, attr)
+}
+
+func testAccOrchestrationBasic(rInt int) string {
+	return fmt.Sprintf(`
+  resource "opc_compute_orchestrated_instance" "test" {
+    name        = "test_orchestration-%d"
+    desired_state = "active"
+		instance {
```

Comment:
Indentation here could use a fix - run these (and other examples) through `terraform fmt`?

Reply:
Good call. Got those for this PR. I'll submit another PR later for fixing the other ones